crates/orchestrator/src/jobs/state_update_job/kzg.rs

```
@@ -0,0 +1,124 @@
+use crate::data_storage::aws_s3::config::AWSS3Config;
+use crate::data_storage::aws_s3::AWSS3;
+use crate::data_storage::{DataStorage, DataStorageConfig};
+use crate::jobs::state_update_job::CURRENT_PATH;
+use alloy::eips::eip4844::BYTES_PER_BLOB;
+use c_kzg::{Blob, Bytes32, KzgCommitment, KzgProof, KzgSettings};
+
+/// Build KZG proof for a given block
+pub async fn build_kzg_proof(
+    block_number: u64,
+    fetch_from_tests: Option<bool>,
+) -> color_eyre::Result<(Vec<u8>, KzgProof)> {
+    let blob_data = fetch_blob_data_for_block(block_number, fetch_from_tests).await?;
+    let mut fixed_size_blob: [u8; BYTES_PER_BLOB] = [0; BYTES_PER_BLOB];
+    fixed_size_blob.copy_from_slice(blob_data.as_slice());
+
+    let x_0_value = fetch_x_0_value_from_os_output(block_number, fetch_from_tests).await?;
+
+    // trusted setup ceremony
+    let trusted_setup_path = CURRENT_PATH.join("src/jobs/state_update_job/trusted_setup.txt");
+    let trusted_setup =
+        KzgSettings::load_trusted_setup_file(trusted_setup_path.as_path()).expect("Error loading trusted setup file");
+
+    let blob = Blob::new(fixed_size_blob);
+    let commitment = KzgCommitment::blob_to_kzg_commitment(&blob, &trusted_setup)?;
+    let (kzg_proof, y_0_value) = KzgProof::compute_kzg_proof(&blob, &x_0_value, &trusted_setup)?;
+
+    // Verifying the proof for double check
+    let eval = KzgProof::verify_kzg_proof(
+        &commitment.to_bytes(),
+        &x_0_value,
+        &y_0_value,
+        &kzg_proof.to_bytes(),
+        &trusted_setup,
+    )?;
+    assert!(eval);
```

this will panic, let's throw an error?